### ISSUE TITLE
posts storage format

### ISSUE THREAD

**Issue by Fil** [vole/vole#67]
This is a point of view on the data storage format.
(v1) is currently a very simple JSON file containing id (message UUID), title (free text), and created (timestamp).
# Content

Vole.cc doesn't have to be limited to exchanging "statuses", but will be used to convey rich data. As a consequence, I think that the content format will have to evolve and gain new possibilities.
#64 mentions geotagging and markdown payload, and I see no reason not to make it open-ended. Obviously we need to establish some common ground, so that different clients can at least display something, even when they don't know what to do with the rich parts.

Maybe http://activitystrea.ms/ could provide a good starting point.

In-Reply-To: and References: are clearly missing.
# Dates

To facilitate tinkering and hacking, dates should be easily readable/writeable by humans. Follow maybe RFC3339 ("1985-04-12T23:20:50.52Z") http://tools.ietf.org/html/rfc3339
# JSON or not

JSON is neat for several reasons, but it's still not as good as plain text when you want to edit it "by hand" or have it indexed by the OS' search engine (spotlight). Some tools such as Jekyll deal with this by using a markdown/textile plain text, with a YAML header (http://jekyllrb.com/docs/frontmatter/). I don't know if it's fundamentally better, but I think it would be a good idea to discuss this.

**Comment by almet** [comment]
Bear in mind that markdown et al. take a lot more time than json files to process (and each client will need to parse these files). 

Having the markdown as the content, and marked as such is otherwise indeed a good idea: we could then write long posts with nice formatting, but I wouldn't make this the default format for processing speed reasons.

**Comment by marktheunissen** [comment]
I agree we should work towards using the http://activitystrea.ms format - I didn't realize it existed and it provides a number of good suggestions.

+1 on the date format.

JSON vs YAML, I don't see a very compelling reason to switch - yes it will be easier to hack the text files if they're YAML front matter and a body, but then we have to parse to JSON to deliver to the browser anyway. Let's see if there's broader interest in this, not totally ruling it out.

Including some parts of Markdown is pretty reasonable, but allowing full formatting including h1-h6 elements and everything else may make the pages look like MySpace. 

We've talked about having pluggable "themes", so that users can switch out their entire front end. Maybe there's a way to allow rich text in the data format, but only display it to those who don't care about the potential design mess. So Vole "base" theme only displays the bare essentials and ignores or strips out all other formatting. Other themes display the formatting in purple blink tags or whatever else people want.

**Comment by Fil** [comment]
To be clear I think the potential for vole goes well beyond the "identi.ca clone" that it looks like atm. It can be a tool to publish full-fledged articles, sites, blogs, podcasts, you name it. medium.com, tumblr, or seenthis.net are all very nice-looking, and are not limited to "a short sentence in plain text".

**Comment by marktheunissen** [comment]
Yes that's true. I would like a rich experience, including longer posts, media, video - we just need to ensure we don't end up with a MySpace. I think that markdown will work for this purpose.

**Comment by jbfavre** [comment]
Working towards  http://activitystrea.ms format would be a great idea imho: this would make Vole's format compatible with other applications like pump.io (statusnet replacement for identi.ca).
That would also allow us to share other people's posts, reply, and so on.

Definitely +1 :)

**Comment by pifantastic** [comment]
Wow, thanks for mentioning activitystrea.ms, I also didn't know of its existence. I'm a +1 on moving towards that.

In regards to Markdown, I think it's ideal, but I don't want to go _any_ farther. Markdown provides a nice subset of HTML, and I think we need to strictly stick to it. Like @marktheunissen said, we do not want to become a myspace. The default Vole theme should remain a very true representation of the converted Markdown, not much different than default browser styles. This way everyone gets content displayed the same, unless they elect into a theme.

It might even be nice to adopt Github flavored markdown for the improved code markup. The task list support doesn't seem too relevant. But as I can imagine early adopters of Vole to be technical, having syntax highlighting could be pretty rad.

**Comment by ANT-Z** [comment]
you can check tent protocol https://tent.io/ i did not study it deep, but it is based on json and may contain many usefull ideas
